src/App.tsx

```
@@ -8,53 +8,182 @@ import { PostsList } from './components/PostsList';
+  const [error, setError] = useState(false);
+  const [errorComments, setErrorComments] = useState(false);
+
+  const [isUserSelected, setIsUserSelected] = useState(false);
+  const [selectedUserId, setSelectedUserId] = useState<number | null>(null);
+
+  const [isSidebarOpen, setIsSidebarOpen] = useState(false);
+
+  const [comments, setComments] = useState<Comment[]>([]);
+
+  async function fetchCreateComment({
+    name,
+    email,
+    body,
+    postId,
+  }: Omit<Comment, 'id'>) {
+    try {
+      const result = await createComments({ name, email, body, postId });
+
+      setComments(currentComments => [...currentComments, result]);
+    } catch {}
+  }
+
+  async function fetchPosts(userId: number) {
+    setSelectedUserId(userId);
+    setIsLoading(true);
+    setError(false);
+
+    if (userId !== selectedUserId) {
+      setIsSidebarOpen(false);
+    }
+
+    setIsUserSelected(!!userId);
+
+    try {
+      const results = await getPosts(userId);
+
+      setPosts(results);
+    } catch {
+      setError(true);
+      setIsLoading(false);
+    } finally {
+      setIsLoading(false);
+    }
+  }
+
+  async function fetchComments(postId: number) {
+    setIsLoadingComments(true);
+    setErrorComments(false);
+
+    try {
+      const results = await getComments(postId);
 
-              <div className="notification is-warning" data-cy="NoPostsYet">
-                No posts yet
+      setComments(results);
+    } catch {
+      setIsLoadingComments(false);
+      setErrorComments(true);
+    } finally {
+      setIsLoadingComments(false);
+    }
+  }
+
+  async function fetchDeleteComment(id: number) {
+    setComments(currentComment =>
+      currentComment.filter(comment => comment.id !== id),
+    );
+
+    try {
+      await deleteComment(id);
+    } catch {}
+  }
+
+  useEffect(() => {
+    const fetchUsers = async () => {
+      try {
+        const loadingUsers = await getUsers();
+
+        setUsers(loadingUsers);
+      } catch {}
```

Empty catch inside useEffect for users fetch hides errors when loading users fails. The app should log the error and/or set an error state so the UI or tests can react.

Suggested fix: `} catch (err) { console.error(err); /* set an error users state if you want to show UI */ }`